docs/julia/Modules.md

```diff
@@ -0,0 +1,182 @@
+        ("$dir/../../../../src/julia/modules/Zygote/ZygoteT.jl", 1e-8),
+        ...
+    ]
+    ```
+1. Add your module and a `tolerance` for the test results to the list.
+    ```julia
+    dir = @__DIR__
+    t_test_implementations = Tuple{String, Float64}[
+        ("$dir/../../../../src/julia/modules/Zygote/ZygoteT.jl", 1e-8),
+        ...
+        ("$dir/../../../../src/julia/modules/YourModule/YourModuleT.jl", absoluteTolerance)
+    ]
+    ```
+
+    `tolerance` is a number used to compare results produced by the current module with the correct results. If an absolute difference between at least one of them exceeds this value then the test is failed.
+
+### Adding a new common test case
+
+Follow these steps to add a new test case for an existing objective that will be shared by all modules:
+
+1. Open `/test/julia/modules/common/TTests.jl` where `T` is the short name of the testing objective.
+2. Find there
+    ```julia
+    @testset "T Module Test ($(basename(module_path)))"
+    ```
+3. In that test set there's a try-finally block that begins with
+    ```julia
+    test = TestLoader.get_t_test(module_name)
+    ```
+    Add your new test case to the end of the try block.
+
+### Adding new objectives
+
+To create a common test suite for a new objective create `/test/julia/modules/common/TTests.jl` with the following content:
+
+```julia
+module TTests
+
+dir = @__DIR__
+t_test_implementations = Tuple{String, Float64}[
+    ("$dir/../../../../src/julia/modules/ModuleName1/ModuleName1T.jl", tolerance1),
+    ...
+]
+
+map!(tup -> (abspath(tup[1]), tup[2]), t_test_implementations, t_test_implementations)
+
+using Test
+include("../../../../src/julia/shared/load.jl")
+include("../../../../src/julia/runner/load.jl")
+include("load.jl")
+import ADPerfTest
+import TestLoader
+using TData
+using TestUtils
+
+@testset "T Module Tests" begin
+
+@testset "T Module Test ($(basename(module_path)))" for (module_path, tolerance) in gmm_test_implementations
+    module_dir, module_filename = splitdir(module_path)
+    module_name, module_ext = splitext(module_filename)
+    # Assert that the path to the module is correct
+    ext_r = @test module_ext == ".jl"
+    ispath_r = @test (ispath(module_path) && !isdir(module_path))
+    if isa(ext_r, Test.Pass) && isa(ispath_r, Test.Pass)
+        need_modify_load_path = !(module_dir ∈ LOAD_PATH)
+        if need_modify_load_path
+            push!(LOAD_PATH, module_dir)
+        end
+        try
+            test = TestLoader.get_t_test(module_name)
+            # Module loads
+            @test isa(test, ADPerfTest.Test{TInput, TOutput})
+            # Test cases go here
+        finally
+            pop!(LOAD_PATH)
+        end
+    end
+end
+
+end
+
+end
+```
+
+This boilerplate code defines a test suite that will be shared by all modules listed in `t_test_implementations`. It includes loading the `ADPerfTest.Test{TInput, TOutput}` object into variable `test.
```

Comment:
Missed \` at the end of the line ("_into variable \`test._")

Reply:
Fixed.